cross-compilation from GNU/Linux/x86_64 to MinGW/i686 broken

Hi,

When the host is GNU/Linux x86_64 and the target is MinGW/i686, then the build system does not detect that this is a cross compilation.

This is because line 230 of `c_check` reads:
``
 $cross = 0 if (($hostarch eq "x86_64") && ($architecture eq "x86"));
``
This test should set `$cross` to 0 only if the previous test on operating system (`$os ne $hostos`) did not already detect cross-compilation.

Best,

Actually I should have been more specific. Building the library succeeds, but then the build process wants to run the tests:
```
OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./sblat1
run-detectors: unable to find an interpreter for ./sblat1
make[1]: *** [Makefile:8: level1] Error 2
```
which of course fails because I can't run Windows binaries on my GNU/Linux box.

I can also verify that the generated `Makefile.conf` does not contain `CROSS=1`.

As explained in my first message, the reason is simply that `c_check` assumes that compiling from x86_64 to x86 is *never* a cross compilation. While in a sense this is true when the host and the target are the same OS, this is not true when cross-compiling from GNU/Linux to MinGW.

--- a/c_check
+++ b/c_check
@@ -223,13 +223,14 @@ $data =~ /globl\s([_\.]*)(.*)/;
 $need_fu      = $1;
 
 $cross = 0;
-$cross = 1 if ($os ne $hostos);
 
 if ($architecture ne $hostarch) {
     $cross = 1;
     $cross = 0 if (($hostarch eq "x86_64") && ($architecture eq "x86"));
     $cross = 0 if (($hostarch eq "mips64") && ($architecture eq "mips"));
 }
+
+$cross = 1 if ($os ne $hostos);
 
 $openmp = "" if $ENV{USE_OPENMP} != 1;
 

